axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/SaleOrderPurchaseServiceImpl.java

```diff
@@ -150,8 +150,8 @@ public PurchaseOrder createPurchaseOrder(
 
     for (SaleOrderLine saleOrderLine : saleOrderLineList) {
       purchaseOrder.addPurchaseOrderLineListItem(
-          purchaseOrderLineServiceSupplychainImpl.createPurchaseOrderLine(
```

Comment:
Never use `Beans.get` inside a loop, as it can cause performance issues. Here replace the class of purchaseOrderLineServiceSupplychainImpl to purchaseOrderLineServiceSupplychain